lib/src/createThemedComponent.js

```diff
@@ -41,32 +40,37 @@ export default function createThemedComponent(C, themedProps = []) {
       this.state = {
         theme,
         props: themedProps.reduce((res, key) => {
-          res[key] = theme.getProp(props[key]);
+          res[key] = this.getThemedPropValue(props[key], theme)
           return res;
         }, {}),
       };
     }
 
+    getThemedPropValue = (prop, theme) => {
+      if (typeof prop == 'object') {
```

Comment:
@smsmdida You are making a gross assumption that any prop of object type is a style, which is not correct. A style can be numeric value (the ones created with StyleSheet.create) and lots of prop values are object (even null is an object).